crates/trie/common/src/hashed_state.rs

```diff
@@ -418,6 +492,92 @@ impl HashedStorageSorted {
     }
 }
 
+/// An iterator that yields chunks of the state updates of at most `size` account and storage
+/// targets.
+///
+/// # Notes
+/// 1. Chunks are expected to be applied in order, because of storage wipes. If applied out of
+///    order, it's possible to wipe more storage than in the original state update.
+/// 2. For each account, chunks with storage updates come first, followed by account updates.
+#[derive(Debug)]
+pub struct ChunkedHashedPostState {
+    flattened: alloc::vec::IntoIter<(B256, FlattenedHashedPostStateItem)>,
+    size: usize,
+}
+
+#[derive(Debug)]
+enum FlattenedHashedPostStateItem {
+    Account(Option<Account>),
+    StorageWipe,
+    StorageUpdate { slot: B256, value: U256 },
+}
+
+impl ChunkedHashedPostState {
+    fn new(hashed_post_state: HashedPostState, size: usize) -> Self {
+        let flattened = hashed_post_state
+            .storages
+            .into_iter()
+            .flat_map(|(address, storage)| {
+                let storage_wipe = if storage.wiped {
+                    itertools::Either::Left(core::iter::once((
+                        address,
+                        FlattenedHashedPostStateItem::StorageWipe,
+                    )))
+                } else {
+                    itertools::Either::Right(core::iter::empty())
+                };
+
+                // Storage wipes should go first
+                storage_wipe.chain(
+                    storage.storage.into_iter().sorted_unstable_by_key(|(slot, _)| *slot).map(
+                        move |(slot, value)| {
+                            (address, FlattenedHashedPostStateItem::StorageUpdate { slot, value })
+                        },
+                    ),
+                )
+            })
+            .chain(hashed_post_state.accounts.into_iter().map(|(address, account)| {
+                (address, FlattenedHashedPostStateItem::Account(account))
+            }))
+            .sorted_by_key(|(address, _)| *address);
```

Comment:
previously, there was a nice comment here about why we don't do unstable, pls add it back